### ISSUE TITLE
Inject node and Debug Nodes unavailable

### ISSUE THREAD

**Issue by SteveRMann** [hassio-addons/addon-node-red#1921]
It is really frustrating when I can’t use inject or debug nodes to test things.

When I open the add-on, the inject nodes don't inject and the debug nodes don't debug.
When I first start Node Red (or redeploy all) the 'inject' buttons are greyed out and unusable. The debug warnings that are normal when I redeploy are also absent.

If other nodes are affected, it isn't apparent. Except that nodes with "Show debug information" selected are also silent.

Running Node-Red from the “Show in Sidebar” does not work.
"Open WebUI" from the add-on configuration page does not work.
Running Node-Red as a Web Page Dashboard **does work**.

I’ve tried everything except the nuclear option.
- Forcing a full deploy
- Different browsers
- Different PC
- Deleting any nodes that give a warning in the logs

## Steps to reproduce
> (How can someone else make/see it happen)

I wish it were that easy.  I've been chasing this for weeks and until opening the add-on in a web page dashboard I could never reproduce a repeatable works/doesn't work.

I have a sandbox installation of Home Assistant that I sometimes use for testing. The sandbox does not exhibit the same problem.

There is at least one [other user ](https://community.home-assistant.io/t/inject-node-not-injecting/754314/13)with the same issue:

**Comment by mshuflin** [comment]
I have the same issue. A workaround is to access the node-red add on by using https://HAIP:1880 (or whatever is set in the network section of the configuration.

I have tried it:
- through the home assistant sidebar using my reverse proxy address - no ability to inject / debug
- through the home assistant sidebar using my internal home assistant address - no ability to inject / debug 
- directly accessing node-red using https://HAIP:1880 - works as expected.

**Comment by ingo916** [comment]
I am having the same issue
through the home assistant sidebar using my internal home assistant address - no ability to inject / debug
directly accessing node-red using https://haip:1880/ - works as expected.

**Comment by taelinn** [comment]
I am having the same issue, running the latest version of the addon

**Comment by sfr0st** [comment]
I am also experiencing this issue. I get the crosshairs when I mouse over the radio button on the inject nodes and the debug nodes aren't showing data in the debug panel. I'm on the latest version on NR, and tried to fall back to the previous version from backup and the issue(s) persisted.

**Comment by taelinn** [comment]
This seems to have resolved itself on my install.

**Comment by mshuflin** [comment]
This is not resolved for me. previously it had temporarily resolved itself after a restart of the host system, but that has not worked recently

**Comment by neilenns** [comment]
I am also facing this issue. I just updated my `node-red-contrib-home-assistant-websocket` package to 0.66.8, and since the restart of node-red and updating my nodes I can't click on my `inject` nodes anymore to test my flows out.

**Comment by smarthomeaf** [comment]
same issue here and confirmed that directly accessing the UI is a valid workaround.   It seems that when this happens, the issue [here](https://github.com/hassio-addons/addon-node-red/issues/1922), also manifest itself.

https://github.com/hassio-addons/addon-node-red/issues/1922

I have permanently solved this in the past, by renaming my nodered config folder.  This will create a fresh install of Nodered and then I just reimport my config.   Not an ideal solution, but if you do not want to access the UI directly, this will be more of a permanent fix (until it happens again a few months later)

**Comment by alexruffell** [comment]
I am having the same issue. In the past it was always due to a Home Assistant server configuration node that got imported with flows I found online. Deleting that configuration node would fix the issue. This time around I don't have any but I found something I'd like others to test.

If I turn on `safe_mode` configuration option and restart NR, the inject nodes and debugs along with all the text under the nodes work again. Doing a full Deploy (as per instructions in pop up) gets you out of `safe_mode` and everything seems to be ok. Anyhow, if I reload the browser window, the issue returns. This doesn't fix the issue but may hopefully help someone who knows the innerworkings fix the issue.

**Comment by SteveRMann** [comment]
Have you tried directly accessing node-red using https://your_home_assistant_ip:1880/

**Comment by alexruffell** [comment]
@SteveRMann Yes, and when I do everything works. I am trying to figure out how to fix the issue when used within HA. It would appear to be a websocket related issue (Chrome shows tons of websocket timeout messages) but all NR flows work fine which puzzles me as I would imagine that those timeouts would cause issues there too.

**Comment by jeremeyi** [comment]
Any updates on this? I am experiencing the same issue?

**Comment by SteveRMann** [comment]
It appears that the problem is in Home Assistant.
Try accessing Node Red directly:
`ha_ip:1880`

**Comment by ikorpershoek** [comment]
I'm not sure if this is a coincidence but I changed my theme to Aurora and everything is working.  I could replicate it on my test server that had the same problem after NR 18.1.1 update.

**Comment by Jurich2004** [comment]
I had the problem described here. I managed to fix it. I deleted the stream imported earlier from another source, then deleted the created entities from the node configuration. Rebooted Node Red and everything worked out. Maybe my case will be useful to someone.

**Comment by sanderlv** [comment]
https://github.com/hassio-addons/addon-node-red/issues/1938#issuecomment-2393313472

**Comment by sanderlv** [comment]
> I had the problem described here. I managed to fix it. I deleted the stream imported earlier from another source, then deleted the created entities from the node configuration. Rebooted Node Red and everything worked out. Maybe my case will be useful to someone.

What stream where?

**Comment by Jurich2004** [comment]
Earlier I downloaded a sample stream from the internet and deployed it. I deleted it and deleted the entities it created. After that everything was fine.

**Comment by Jurich2004** [comment]
<img width="206" alt="ex" src="https://github.com/user-attachments/assets/8ed7eaa5-9fa6-47de-82d6-78862e30d2b9">

**Comment by eMeF1** [comment]
I'm having the same issue with the inject nodes not working and no labels available. I remember importing some flow around the time the issue first appeared, but I've already deleted it without any impact. Today, I removed the duplicated HA server configuration, and after deploying, it started to work again. However, it didn't survive a page reload. Interestingly, if I leave the page open, the issue disappears after around 10 minutes. I hope this information is helpful for anyone looking into the problem.

**Comment by alexruffell** [comment]
Good news! I just updated `node-red-contrib-home-assistant-websocket` to `0.74.2` from `Manage Palette` and everything works again!

**Comment by alexio70** [comment]
@alexruffell you made my day; works perfectly !!

**Comment by alexruffell** [comment]
The issue, or a very similar one, has returned :( My entities aren't listed in nodes and sometimes there is no target field. Inject and debug buttons work but debug output sometimes requires me to use NR via port 1880. Any others having these issues?

**Comment by alvinchen1** [comment]
This issue might be resolved in 19.0.0 released today.  Please try the latest update and see https://github.com/hassio-addons/addon-node-red/issues/1986#issuecomment-2571378011

**Comment by frenck** [comment]
Yup! Fixed by https://github.com/hassio-addons/addon-node-red/releases/tag/v19.0.0
